interfaces/builtin/tpm.go

```diff
@@ -0,0 +1,67 @@
+package builtin
+
+import "github.com/snapcore/snapd/interfaces"
+
+var tpmPermanentSlotAppArmor = []byte(`
+# Description: for those who need to talk to the system TPM chip over /dev/tpm0
+# Usage: common
+
+/dev/tpm0 rw,
+`)
+
+type TpmInterface struct { }
+
+func (iface *TpmInterface) Name() string {
+	return "tpm"
+}
+
+func (iface *TpmInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus, interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *TpmInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityUDev, interfaces.SecurityDBus:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *TpmInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return tpmPermanentSlotAppArmor, nil
+	case interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev:
+		return nil, nil
```

Comment:
Based on your comment, I think you are saying that access to /dev/tpm0 would use standard syscalls and anything beyond that is simply because the software opening it has additional functionality that is correctly covered by network and network-bind.

What Simon and I want to make sure is that if the tpm snap legitimately needs network (ie, it reaches out over the network) and network-bind (it opens a network port), then, sure, don't worry about extra syscalls, but if you are using those interfaces when the tpm snap doesn't reach out over the network or bind to a port, then the needed syscalls should be here too.